### PR TITLE
Fix broken test in ndslicing

### PR DESCRIPTION
**Pull request by mwcraig**
The former comparison was identified in astropy/astropy#14920 as a comparison that wasn't tested. "is" was not the appropriate comparison anyway, and comparing a dict to an UnknownUncertainty was also not sensible.

### Description

This pull request is to address a (failing) comparison in the `NDSlicing` tests that
was quietly failing.

Addresses the `nddata` part of #14920.

### CODE REVIEW

**Review by eerovaher — CHANGES_REQUESTED** [review]
In #14919 the [rule B015](https://beta.ruff.rs/docs/rules/#flake8-bugbear-b) was removed from the global Ruff ignore list and was added to the `nddata` sub-package ignore list because of the comparison this pull request addresses, so `.ruff.roml` also needs to be edited so that Ruff will enforce B015 in `nddata`: https://github.com/astropy/astropy/blob/0a71105fbaa71439206d496a44df11abc0e3ac96/.ruff.toml#L297-L298

**Comment by mwcraig** [comment]
@eerovaher -- I *think* I correctly change the ruff config file.

@eerovaher and @pllim -- if this looks good please don't merge it. I'm meeting with @bmorris3 this afternoon and would like to use this PR to walk him through his first merge as an `nddata` maintainer.

**Comment by pllim** [comment]
There is conflict, need a rebase. Good luck!

**Comment by mwcraig** [comment]
Now it is actually fixed, I think 😬

**Comment by mwcraig** [comment]
pre-commit.ci autofix

**Review by bmorris3 — APPROVED** [review]
Looks good to me.

**Comment by pllim** [comment]
@mwcraig / @bmorris3 -- Also an excellent opportunity to try your hand at manual backports. 😉

**Comment by pllim** [comment]
p.s. Probably caused by Ruff stuff since we don't backport those.

**Comment by mwcraig** [comment]
> @mwcraig / @bmorris3 -- Also an excellent opportunity to try your hand at manual backports. 😉 

Ok, can look in the morning

**Comment by mwcraig** [comment]
This is getting backported to the 5.3.x branch, not 5.0.x, right?

**Comment by mwcraig** [comment]
Also, should I just backport the test change or also the ruff configuration?

**Comment by pllim** [comment]
@mwcraig 

> This is getting backported to the 5.3.x branch, not 5.0.x, right?

Yes, just 5.3.x because this is not really a "critical bug fix that we must put in LTS", right?

> should I just backport the test change or also the ruff configuration?

There is no need to backport ruff. I actually think it is the ruff stuff that is causing conflict, not the actual fix.

**Pull request merged**
